Modules/Brand/Http/Services/BrandService.php

```diff
@@ -0,0 +1,287 @@
+    {
+
+        $brands = User::where('role', '=', ROLE::ROLE_BRAND)
+            ->whereIn('id', function ($query) {
+                $query->select('user_id')
+                    ->from('brands');
+            })
+            ->with('profile');
+        $filteredCount = $brands->count();
+        $start = $request->start ?? 0;
+        $length = $request->length ?? 0;
+        $brands = $brands->offset($start)->limit($length);
+        $result = DataTables::of($brands)
+            ->addIndexColumn()
+            ->addColumn('brand_name', function ($row) {
+                return $row->profile->brand_name;
+            })
+            ->addColumn('go_live', function ($row) {
+                return $row->profile->go_live;
+            })
+            ->addColumn('phone_number', function ($row) {
+                return $row->profile->phone_number;
+            })
+            ->addColumn('action', function ($row) {
+                return '<a href="'.url('/brand/details/' . $row->id).'"><span class="fas fa-eye"></span></a> <a href="'.url('/product/' . $row->id).'"><span class="fas fa-list"></span></a>  <a href="javascript:void(0)" onclick="event.preventDefault(); confirmDelete('.$row->id.');"><span class="fas fa-trash"></span></a> ';
+            })
+            ->rawColumns(['action'])
+            ->make(true);
+        $resultData = $result->getData(true);
+        $resultData['recordsFiltered'] = $filteredCount;
+        $resultData['draw'] = $request->draw;
+
+        return $resultData;
+    }
+
+    /**
+     * Details of brand
+     *
+     * @param $id
+     * @return mixed
+     */
+    public function detail($id): mixed
+    {
+
+        $brand = User::findOrFail($id);
+        $detail = $brand->profile;
+        $category = Category::where('id',$detail->prime_cat)->first();
+        $headQuarter = Country::where('id',$detail->headquatered)->first();
+        $productMade = Country::where('id',$detail->product_made)->first();
+        $productShipped = Country::where('id',$detail->product_shipped)->first();
+
+        $detail['name'] = $brand->first_name.' '.$brand->last_name;
+        $detail['email'] = $brand->email ?? '';
+        $detail['prime-category'] = $category->title ?? '';
+        $detail['head-quarter'] = $headQuarter->name ?? '';
+        $detail['product-made'] = $productMade->name ?? '';
+        $detail['product-shipped'] = $productShipped->name ?? '';
+
+        return $detail;
+    }
+
+    /**
+     * Delete Brand
+     *
+     * @param $id
+     * @return array
+     */
+    public function delete($id): array
+    {
+        $user = User::findOrFail($id);
+        $user->delete();
+        return [
+            'res' => true,
+            'msg' => 'Successfully deleted',
+            'data' => ''
+        ];
+    }
+
+    /**
+     * Get all country
+     *
+     * @return array
+     */
+    public function getCountry(): array
+    {
+        $countries = Country::orderBy('name', 'ASC')->get();
```

Comment:
when you query for a model and you need only some data, then get only required one, you can use pluck or select method to get the required data only.
$countries = Country::orderBy('name', 'ASC')->pluck('name', 'id', 'shortname', 'phonecode');
This way it will not impact much on the DB.

Reply:
Fixed